iceoryx_binding_c/source/c_listener.cpp

```diff
@@ -203,3 +204,51 @@ void iox_listener_detach_client_event(iox_listener_t const self,
 
     self->detachEvent(*client, c2cpp::clientEvent(clientEvent));
 }
+
+iox_ListenerResult
+iox_listener_attach_service_discovery_event(iox_listener_t const self,
+                                            iox_service_discovery_t const serviceDiscovery,
+                                            const ENUM iox_ServiceDiscoveryEvent serviceDiscoveryEvent,
+                                            void (*callback)(iox_service_discovery_t))
+{
+    iox::cxx::Expects(self != nullptr);
```

Comment:
When the `contextData != nullptr` requirement is removed one could call
```
return iox_listener_attach_service_discovery_event_with_context_data(self, serviceDiscovery, serviceDiscoveryEvent, callback, nullptr);
```
and remove all the code.

Reply:
Same here.